vault/token_store.go

```diff
@@ -3061,15 +3063,13 @@ func (ts *TokenStore) tokenStoreRoleCreateUpdate(ctx context.Context, req *logic
 	pathSuffixInt, ok := data.GetOk("path_suffix")
 	if ok {
 		pathSuffix := pathSuffixInt.(string)
-		if pathSuffix != "" {
-			matched := pathSuffixSanitize.MatchString(pathSuffix)
-			if !matched {
-				return logical.ErrorResponse(fmt.Sprintf(
-					"given role path suffix contains invalid characters; must match %s",
-					pathSuffixSanitize.String())), nil
-			}
-			entry.PathSuffix = pathSuffix
+		matched := pathSuffixSanitize.MatchString(pathSuffix)
```

Comment:
I think the logic would be a little cleaner to follow here if you used a switch so that the base case of `""` is handled up front, but it's not a big deal.

Reply:
Thanks. You are right, it's not really clean.